packages/core/src/external/carequality/ihe-gateway-v2/saml/saml-client.ts

```diff
@@ -76,15 +78,24 @@ export async function sendSignedXml({
     secureOptions: constants.SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION,
   });
 
-  const response = await axios.post(url, signedXml, {
-    timeout: 120000,
-    headers: {
-      "Content-Type": "application/soap+xml;charset=UTF-8",
-      Accept: "application/soap+xml",
-      "Cache-Control": "no-cache",
+  const response = await executeWithNetworkRetries(
+    async () => {
+      return axios.post(url, signedXml, {
+        timeout: 120000,
+        headers: {
+          "Content-Type": "application/soap+xml;charset=UTF-8",
+          Accept: "application/soap+xml",
+          "Cache-Control": "no-cache",
+        },
+        httpsAgent: agent,
+      });
     },
-    httpsAgent: agent,
-  });
+    {
+      initialDelay: 3000,
+      maxAttempts: 3,
+      httpCodesToRetry: ["ECONNREFUSED", "ECONNRESET", "ETIMEDOUT"],
```

Comment:
This PR introduces retry on timeout without the need to specify the Axios/Http Code for it: https://github.com/metriport/metriport/pull/2285/files#diff-8db132fb67be37a37c157aaf0a89e0dddc21ac2d9b8b54c7674a35a4ea34174fR13

Can leave as is, rebase, or just copy the code of `net/retry.ts`, up to you. I'd just ask to add a comment/TODO if we leave as is so next time we touch this we can simplify it a bit.

Reply:
I am leaving as is and adding a todo!